BiometricsFingerprint.cpp

```diff
@@ -599,10 +609,13 @@ void * BiometricsFingerprint::worker_thread(void *args){
                         thisPtr->mClientCallback->onAuthenticated(devId, fid, gid, hidl_vec<uint8_t>());
                     }
                 } else {
-                    setState(sdev, STATE_IDLE);
-                    thisPtr->mClientCallback->onError(devId, FingerprintError::ERROR_CANCELED, 0);
-                    raise(SIGKILL);
-                    break;
+                      /*
```

Comment:
This stuff *really* does not belong in auth. It should rather return a critical error to the thread loop, and then that should reinitialize.

Reply:
I actually agree with you, completely, on this....
Though, that was the first thing I tried and it didn't work as expected (when we were talking about it, that was the way I was trying....)

Another thing to notice is that if we return an error to Android, it will cancel the authentication loop until you put display back on or put display on-off!